tox.ini

```
@@ -179,3 +179,30 @@ install_command = pip install --no-deps {opts} {packages}
 commands =
     conda list
     {env:PYTEST_COMMAND} {posargs}
+
+[testenv:pyinstaller]
+description = check that sunpy can be included in a pyinstaller bundle
+changedir = .pyinstaller
+deps =
+    # The developer version of PyInstaller doesn't work yet because it
+    # requires https://github.com/pyinstaller/pyinstaller/pull/4888 -
+    # so for now we pin to the latest working commit.
+    git+https://github.com/pyinstaller/pyinstaller.git@c111ec5c#egg=PyInstaller
+    # PyInstaller is not yet compatible with Matplotlib 3.3 so we pin
+    # Matplotlib to an older version for now - see the following issue
+    # for more details: https://github.com/pyinstaller/pyinstaller/issues/5004
+    matplotlib<3.3
+extras =
+    tests
+    net
+    all
+commands =
+    pyinstaller --onefile run_sunpy_tests.py \
+                --distpath . \
+                --additional-hooks-dir hooks \
+                --exclude-module tkinter \
+                --hidden-import pytest \
+                --hidden-import pytest_doctestplus.plugin \
+                --hidden-import pytest_mpl.plugin
+
+    ./run_sunpy_tests
```

Rather than making a separate script can we not do this?

```suggestion
    python -m pytest -p no:warnings --doctest-rst -m "not mpl_image_compare" --pyargs sunpy
```

---

Correct me if I'm wrong, but we're creating the executable and making it run all the rests right (to ensure that the executable created for sunpy is working properly)?  What you've suggested would just execute the python statement in the tox environment, which is essentially just testing normally.